plugin/pkg/admission/limitranger/admission.go

```diff
@@ -210,76 +210,107 @@ func NewLimitRanger(actions LimitRangerActions) (*LimitRanger, error) {
 	setLimits := []string{}
-	if container.Resources.Limits == nil {
-		container.Resources.Limits = api.ResourceList{}
-	}
-	if container.Resources.Requests == nil {
-		container.Resources.Requests = api.ResourceList{}
-	}
+
 	for k, v := range defaultRequirements.Limits {
-		_, found := container.Resources.Limits[k]
+		if resources.Limits == nil {
+			resources.Limits = api.ResourceList{}
+		}
+
+		_, found := resources.Limits[k]
 		if !found {
-			container.Resources.Limits[k] = v.DeepCopy()
+			resources.Limits[k] = v.DeepCopy()
 			setLimits = append(setLimits, string(k))
 		}
 	}
 	for k, v := range defaultRequirements.Requests {
-		_, found := container.Resources.Requests[k]
+		if resources.Requests == nil {
+			resources.Requests = api.ResourceList{}
+		}
+
+		_, found := resources.Requests[k]
 		if !found {
-			container.Resources.Requests[k] = v.DeepCopy()
+			resources.Requests[k] = v.DeepCopy()
 			setRequests = append(setRequests, string(k))
 		}
 	}
 	if len(setRequests) > 0 {
 		sort.Strings(setRequests)
-		a := strings.Join(setRequests, ", ") + fmt.Sprintf(" request for %s %s", annotationPrefix, container.Name)
+		a := strings.Join(setRequests, ", ") + fmt.Sprintf(" request for %s %s", annotationPrefix, name)
 		annotations = append(annotations, a)
 	}
 	if len(setLimits) > 0 {
 		sort.Strings(setLimits)
-		a := strings.Join(setLimits, ", ") + fmt.Sprintf(" limit for %s %s", annotationPrefix, container.Name)
+		a := strings.Join(setLimits, ", ") + fmt.Sprintf(" limit for %s %s", annotationPrefix, name)
 		annotations = append(annotations, a)
 	}
 	return annotations
 }
 
 // mergePodResourceRequirements merges enumerated requirements with default requirements
 // it annotates the pod with information about what requirements were modified
-func mergePodResourceRequirements(pod *api.Pod, defaultRequirements *api.ResourceRequirements) {
+func mergePodResourceRequirements(pod *api.Pod, defaultResources *defaultResources, podLevelResourcesEnabled bool) {
 	annotations := []string{}
+	if podLevelResourcesEnabled && (defaultResources.pod.Requests != nil || defaultResources.pod.Limits != nil) {
+		if pod.Spec.Resources == nil {
+			pod.Spec.Resources = &api.ResourceRequirements{}
+		}
+
+		annotations = mergeResources(pod.Name, pod.Spec.Resources, &defaultResources.pod, "pod", annotations)
+	}
 
 	for i := range pod.Spec.Containers {
-		annotations = mergeContainerResources(&pod.Spec.Containers[i], defaultRequirements, "container", annotations)
+		annotations = mergeResources(pod.Spec.Containers[i].Name, &pod.Spec.Containers[i].Resources, &defaultResources.container, "container", annotations)
 	}
 
 	for i := range pod.Spec.InitContainers {
-		annotations = mergeContainerResources(&pod.Spec.InitContainers[i], defaultRequirements, "init container", annotations)
+		annotations = mergeResources(pod.Spec.Containers[i].Name, &pod.Spec.InitContainers[i].Resources, &defaultResources.container, "init container", annotations)
```

Comment:
Should it be `pod.Spec.InitContainers[i].Name`

Reply:
Done, thank you.